### ISSUE TITLE
📝 HTML formatter collapses inline element content onto single line in Svelte files

### ISSUE THREAD

**Issue by petrzelka** [biomejs/biome#9099]
### Environment information

When formatting Svelte files with experimentalFullSupportEnabled, the HTML formatter collapses inline element content (like <span> or custom components) with embedded expressions onto a single line, instead of keeping them on separate lines.
This affects both css and strict whitespace sensitivity modes, producing different but equally incorrect results.
Input
```
 <ul>
            <li>
              <div>
              
                {name}

              
              </div>
            </li>
        </ul>
```
Expected output (matches Prettier with htmlWhitespaceSensitivity: "css")
```
<ul>
  <li>
    <div>
      {name}
    </div>
  </li>
</ul>

```
Actual output
With whitespaceSensitivity: "css" (default):
```
<ul>
	<li>
		<div>{name}</div>
	</li>
</ul>
```
With whitespaceSensitivity: "strict":
```
<ul>
	<li> <div> {name} </div> </li>
</ul>
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
It seems that prettier will preserve the newline if it is present, but will not introduce a newline if it does not exist. This playground shows it a bit better: https://biomejs.dev/playground/?code=IAA8AHUAbAA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAPABsAGkAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AGQAaQB2AD4ACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHsAbgBhAG0AZQB9ADwAIQAtAC0AIABuAGUAdwBsAGkAbgBlACAAcAByAGUAcwBlAHIAdgBlAGQAIAAtAC0APgAKAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwALwBkAGkAdgA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAPAAvAGwAaQA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAPABsAGkAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AGQAaQB2AD4AewBuAGEAbQBlAH0APAAhAC0ALQAgAG4AZQB3AGwAaQBuAGUAIABuAG8AdAAgAGkAbgB0AHIAbwBkAHUAYwBlAGQAIAAtAC0APgAKAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwALwBkAGkAdgA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAPAAvAGwAaQA%2BAAoAIAAgACAAIAAgACAAIAAgADwALwB1AGwAPgAKAAoA&language=svelte

**Comment by vladshcherbin** [comment]
Same thing happens with even simple `<svelte:head>` element, from:

```svelte
<svelte:head>
  <title>Yolo</title>
</svelte:head>
```

biome formats to:

```svelte
<svelte:head> <title>Yolo</title> </svelte:head>
```

[Playground Reproduction](https://biomejs.dev/playground/?code=PABzAHYAZQBsAHQAZQA6AGgAZQBhAGQAPgAKACAAIAA8AHQAaQB0AGwAZQA%2BAFkAbwBsAG8APAAvAHQAaQB0AGwAZQA%2BAAoAPAAvAHMAdgBlAGwAdABlADoAaABlAGEAZAA%2BAAoA&language=svelte)

**Comment by gameroman** [comment]
Same with Astro

```astro
<Layout {title}>
  <h1>{title}</h1>
</Layout>
```

to

```astro
<Layout {title}> <h1>{title}</h1> </Layout>
```

**Comment by tryforceful** [comment]
FWIW it looks like this might be a regression even for the whole HTML formatter, not just Svelte? I'm using Biome with Vue and I have the same issue.

See this reproduction with the language chosen as `HTML`, not even `Vue` or `Svelte`:

https://biomejs.dev/playground/?enabledLinting=false&enabledAssist=false&code=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8AHMAbABvAHQAPgAKACAAIAAgACAAPABhAD4AVABlAHMAdAA8AC8AYQA%2BAAoAIAAgADwALwBzAGwAbwB0AD4ACgAgACAAPAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAAgACAAPABNAHkASQBjAG8AbgAgAC8APgAKACAAIAA8AC8AdABlAG0AcABsAGEAdABlAD4ACgAgACAAPABNAHkAQwB1AHMAdABvAG0AQwBvAG0AcABvAG4AZQBuAHQAIABzAGkAegBlAD0AIgB4AHMAIgA%2BAAoAIAAgACAAIAA8AEMAaABpAGwAZABDAG8AbQBwAG8AbgBlAG4AdAAgAC8APgAKACAAIAA8AC8ATQB5AEMAdQBzAHQAbwBtAEMAbwBtAHAAbwBuAGUAbgB0AD4ACgAgACAAPABkAGkAdgA%2BAAoAIAAgACAAIAA8AHMAcABhAG4APgAKACAAIAAgACAAIAAgADwAcwBlAGMAdABpAG8AbgA%2BAAoAIAAgACAAIAAgACAAIAAgADwAYQByAHQAaQBjAGwAZQA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAA8AGkAbgBwAHUAdAAgAC8APgAKACAAIAAgACAAIAAgACAAIAA8AC8AYQByAHQAaQBjAGwAZQA%2BAAoAIAAgACAAIAAgACAAPAAvAHMAZQBjAHQAaQBvAG4APgAKACAAIAAgACAAPAAvAHMAcABhAG4APgAKACAAIAA8AC8AZABpAHYAPgAKADwALwB0AGUAbQBwAGwAYQB0AGUAPgA%3D&language=html

I would expect the output to be the same as Prettier here. In earlier patch versions of 2.3 it does work this way. Somewhere in 2.3.x to 2.4 it broke this way.

<img width="1213" height="1095" alt="Image" src="https://github.com/user-attachments/assets/751d5c29-5461-477f-b271-443b4aa27ab5" />

**Comment by dyc3** [comment]
Seems likely @tryforceful. Would you mind fixing your playground link? it doesn't show what you have in your screenshot

**Comment by tryforceful** [comment]
Sorry, updated my link above! 🙏

**Comment by ematipico** [comment]
Closing. The issues has been fixed in one of latest versions of Biome.

**Comment by vladshcherbin** [comment]
@ematipico doesn't look to be fixed in [latest playground](https://biomejs.dev/playground/?whitespaceSensitivity=strict&code=PABzAHYAZQBsAHQAZQA6AGgAZQBhAGQAPgAKACAAIAA8AHQAaQB0AGwAZQA%2BAFkAbwBsAG8APAAvAHQAaQB0AGwAZQA%2BAAoAPAAvAHMAdgBlAGwAdABlADoAaABlAGEAZAA%2BAA%3D%3D&language=svelte)

```svelte
<svelte:head>
  <title>Yolo</title>
</svelte:head>
```

<img width="1213" height="561" alt="Image" src="https://github.com/user-attachments/assets/fbbd6362-a057-4dd2-9344-9b9d31095d32" />

**Comment by petrzelka** [comment]
not fixed in 2.4.9 that was just released - is it scheduled for a later release?

**Comment by ematipico** [comment]
Thank you for checking in. Re-opening

**Comment by yefu24324** [comment]
Encountering the same problem

@biomejs/biome@2.4.12

## Source

```vue
<template>
  <router-view v-slot="{ Component }" ref="view">
    <transition name="fade-transform" mode="out-in">
      <component :is="Component"></component>
    </transition>
  </router-view>
  <div id="app">
    <div id="app"></div>
  </div>
  <training id="app">
    <training id="app"></training>
  </training>
</template>
```

## Expected Output

```vue
<template>
  <router-view v-slot="{ Component }" ref="view">
    <transition name="fade-transform" mode="out-in">
      <component :is="Component"></component>
    </transition>
  </router-view>
  <div id="app">
    <div id="app"></div>
  </div>
  <training id="app">
    <training id="app"></training>
  </training>
</template>
```

## Actual Output

```vue
<template>
  <router-view v-slot="{ Component }" ref="view">
    <transition name="fade-transform" mode="out-in"> <component :is="Component"></component> </transition>
  </router-view>
  <div id="app">
    <div id="app"></div>
  </div>
  <training id="app"> <training id="app"></training> </training>
</template>
```

## Biome Playground

The behavior of the playground is inconsistent with the local biome

https://biomejs.dev/playground/?whitespaceSensitivity=strict&tab=formatter&pane=Diagnostics&code=PAB0AGUAbQBwAGwAYQB0AGUAPgAKACAAIAA8AHIAbwB1AHQAZQByAC0AdgBpAGUAdwAgAHYALQBzAGwAbwB0AD0AIgB7ACAAQwBvAG0AcABvAG4AZQBuAHQAIAB9ACIAIAByAGUAZgA9ACIAdgBpAGUAdwAiAD4ACgAgACAAIAAgADwAdAByAGEAbgBzAGkAdABpAG8AbgAgAG4AYQBtAGUAPQAiAGYAYQBkAGUALQB0AHIAYQBuAHMAZgBvAHIAbQAiACAAbQBvAGQAZQA9ACIAbwB1AHQALQBpAG4AIgA%2BAAoAIAAgACAAIAAgACAAPABjAG8AbQBwAG8AbgBlAG4AdAAgADoAaQBzAD0AIgBDAG8AbQBwAG8AbgBlAG4AdAAiAD4APAAvAGMAbwBtAHAAbwBuAGUAbgB0AD4ACgAgACAAIAAgADwALwB0AHIAYQBuAHMAaQB0AGkAbwBuAD4ACgAgACAAPAAvAHIAbwB1AHQAZQByAC0AdgBpAGUAdwA%2BAAoAIAAgADwAZABpAHYAIABpAGQAPQAiAGEAcABwACIAPgAKACAAIAAgACAAPABkAGkAdgAgAGkAZAA9ACIAYQBwAHAAIgA%2BADwALwBkAGkAdgA%2BAAoAIAAgADwALwBkAGkAdgA%2BAAoAIAAgADwAdAByAGEAaQBuAGkAbgBnACAAaQBkAD0AIgBhAHAAcAAiAD4ACgAgACAAIAAgADwAdAByAGEAaQBuAGkAbgBnACAAaQBkAD0AIgBhAHAAcAAiAD4APAAvAHQAcgBhAGkAbgBpAG4AZwA%2BAAoAIAAgADwALwB0AHIAYQBpAG4AaQBuAGcAPgAKADwALwB0AGUAbQBwAGwAYQB0AGUAPgA%3D&language=vue
